cmd/dependabot/internal/cmd/update.go

```diff
@@ -156,6 +85,86 @@ var updateCmd = &cobra.Command{
 	},
 }
 
+func extractInput(cmd *cobra.Command) (*model.Input, []byte, error) {
+	var input *model.Input
+	var inputRaw []byte
+
+	if file != "" {
+		if len(cmd.Flags().Args()) > 0 {
+			return nil, nil, errors.New("cannot use file and arguments together")
```

Comment:
This makes sense, but seems a little buried, it'd be easy to overlook if you're down in the cmd.flags parsing section later.

I wonder if restructuring it might make things a little more intuitive... one possible solution:

1. what types of input are present?
2. error checking that no invalid combos of input are present, maybe use a pseudo-switch case for readability
3. for allowed combos of input, if they have overlapping input resulting in conflicts, resolve the input precedence
4. return the input

That way when working within a particular input processing section, you don't have to also remember to defend against input happening from other vectors... 

And if you're skimming this code trying to figure out what "wins" in a given scenario, it's obvious w/o the distraction of the input parsing code.